src/v/iceberg/table_metadata.h

```diff
@@ -85,6 +85,24 @@ struct table_metadata {
     // TODO: (optional) statistics
     // TODO: (optional) partition_statistics
 
+    const partition_spec* get_partition_spec(partition_spec::id_t id) const {
+        for (const auto& spec : partition_specs) {
+            if (spec.spec_id == id) {
+                return &spec;
+            }
+        }
+        return nullptr;
+    }
+
+    const schema* get_schema(schema::id_t id) const {
+        for (const auto& s : schemas) {
+            if (s.schema_id == id) {
+                return &s;
+            }
+        }
+        return nullptr;
+    }
```

Comment:
Just FYI these can be done with `std::ranges::find(schemas, id, &schema::id)`

Reply:
nice, though I am a bit of a plain for loop enjoyer :)

Reply:
Heh, usually I'm the same. In this case the predicate is super simple so it's nice and easy to read